### PR TITLE
User items

### PR DESCRIPTION
**Pull request by EngrTaofeek**
Requested changes from previous pull request #674

### CODE REVIEW

File: app/src/main/java/org/greenstand/android/TreeTracker/models/user/User.kt [hunk 1]
```diff
@@ -5,6 +5,8 @@ data class User(
     val wallet: String,
     val firstName: String,
     val lastName: String?,
+    val email: String?,
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/models/user/User.kt:8]: There is no reason to add these fields. Please remove.

File: app/src/main/java/org/greenstand/android/TreeTracker/models/user/Users.kt [hunk 2]
```diff
@@ -135,6 +135,8 @@ class Users(
             wallet = planterInfoEntity.identifier,
             firstName = planterInfoEntity.firstName,
             lastName = planterInfoEntity.lastName,
+            email = planterInfoEntity.email,
+            phone = planterInfoEntity.phone,
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/models/user/Users.kt:139]: Not needed, see above.

File: app/src/main/java/org/greenstand/android/TreeTracker/camera/SelfieScreen.kt [hunk 3]
```diff
@@ -21,6 +24,9 @@ import org.greenstand.android.TreeTracker.view.*
 fun SelfieScreen() {
     val navController = LocalNavHostController.current
     val cameraControl = remember { CameraControl() }
+    val displayMetrics = Resources.getSystem().displayMetrics
+    val dpWidth = (displayMetrics.widthPixels / displayMetrics.density + 0.5).toInt()
+
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/camera/SelfieScreen.kt:29]: Please remove this. The code is not used anywhere.

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/NameEntryView.kt [hunk 4]
```diff
@@ -77,7 +80,7 @@ fun NameEntryView(
                 rightAction = {
                     ArrowButton(
                         isLeft = false,
-                        isEnabled = state.name != null
+                        isEnabled = (state.name != null && state.organization != null)
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/signup/NameEntryView.kt:83]: The organization can be left null. Only the name is required for now.

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/NameEntryView.kt [hunk 5]
```diff
@@ -117,7 +120,9 @@ fun NameEntryView(
                 ),
                 keyboardActions = KeyboardActions(
                     onGo = {
-                        cameraLauncher.launch(true)
+                        if (state.name != null && state.organization != null) {
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/signup/NameEntryView.kt:123]: Same as above.

File: app/src/main/java/org/greenstand/android/TreeTracker/view/SelectableItemDetail.kt [hunk 6]
```diff
@@ -84,48 +84,8 @@ fun UserButton(
...
-@Composable
-fun WalletUserButton(
-    user: User,
-    isSelected: Boolean,
-    buttonColors: DepthButtonColors,
-    selectedColor: Color,
-    onClick: () -> Unit
-) {
+    val credential: String? = if (user.email.isNullOrEmpty()) user.phone else user.email
+
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/view/SelectableItemDetail.kt:88]: Let's change this to use the wallet. The wallet field can be filled in the User object from the identifier field in the db object.

**Review by Elforama — CHANGES_REQUESTED** [review]
Please address all the comments. Otherwise, looks good.

File: app/src/main/java/org/greenstand/android/TreeTracker/walletselect/WalletSelectScreen.kt [hunk 7]
```diff
@@ -30,12 +30,7 @@ import org.greenstand.android.TreeTracker.models.NavRoute
 import org.greenstand.android.TreeTracker.models.user.User
 import org.greenstand.android.TreeTracker.root.LocalNavHostController
 import org.greenstand.android.TreeTracker.root.LocalViewModelFactory
-import org.greenstand.android.TreeTracker.view.ActionBar
-import org.greenstand.android.TreeTracker.view.AppButtonColors
-import org.greenstand.android.TreeTracker.view.AppColors
-import org.greenstand.android.TreeTracker.view.ArrowButton
-import org.greenstand.android.TreeTracker.view.LocalImage
-import org.greenstand.android.TreeTracker.view.WalletUserButton
+import org.greenstand.android.TreeTracker.view.*
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/walletselect/WalletSelectScreen.kt:33]: We don't want wildcard imports. Did this help is pass CI/CD? If it builds, it should be able to pass those tests.

Comment (EngrTaofeek) [reply app/src/main/java/org/greenstand/android/TreeTracker/walletselect/WalletSelectScreen.kt:33]: It was automatic, I didn't even notice. Is that all that is left for this?? I want to make a pull request for permission management at runtime, just a few things left to tidy up on that branch.

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt [hunk 8]
```diff
@@ -59,17 +65,24 @@ class SignupViewModel(private val users: Users) : ViewModel() {
 
     fun updateEmail(email: String) {
         // TODO validate data and show errors if needed, after click
-        _state.value = _state.value?.copy(credential = Credential.Email().apply { text = email })
+        _state.value = _state.value?.copy(email = email)
+        _state.value = _state.value?.copy(isEmailValid = Validation.isEmailValid(email))
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt:69]: For all the state updates, please update all params with a single copy method.

For example:
_state.value = _state.value?.copy(param1 = 1, param2 = 2, etc..)

Without that, you'll be updating the state once per each value.

**Review by Elforama — APPROVED** [review]
Looks good, the remaining comments can be addressed later.

File: app/src/main/java/org/greenstand/android/TreeTracker/walletselect/WalletSelectScreen.kt (hunk 7 above)
Comment (Elforama) [reply app/src/main/java/org/greenstand/android/TreeTracker/walletselect/WalletSelectScreen.kt:33]: Please pay close attention to your PRs.

**Pull request merged**